pkg/controller/podautoscaler/horizontal.go

```diff
@@ -148,22 +148,41 @@ func (a *HorizontalController) computeReplicasForCPUUtilization(hpa *autoscaling
 		a.eventRecorder.Event(hpa, api.EventTypeWarning, "InvalidSelector", errMsg)
 		return 0, nil, time.Time{}, fmt.Errorf(errMsg)
 	}
-	currentUtilization, timestamp, err := a.metricsClient.GetCPUUtilization(hpa.Namespace, selector)
+	utilizationInfo, err := a.metricsClient.GetCPUUtilization(hpa.Namespace, selector)
 
 	// TODO: what to do on partial errors (like metrics obtained for 75% of pods).
 	if err != nil {
 		a.eventRecorder.Event(hpa, api.EventTypeWarning, "FailedGetMetrics", err.Error())
 		return 0, nil, time.Time{}, fmt.Errorf("failed to get CPU utilization: %v", err)
 	}
 
-	utilization := int32(*currentUtilization)
+	utilization := int32(utilizationInfo.Utilization)
 
-	usageRatio := float64(utilization) / float64(targetUtilization)
-	if math.Abs(1.0-usageRatio) > tolerance {
-		return int32(math.Ceil(usageRatio * float64(currentReplicas))), &utilization, timestamp, nil
+	usageRatio := utilizationInfo.Utilization / float64(targetUtilization)
+	if math.Abs(1.0-usageRatio) <= tolerance {
+		// if the change was too small, ignore it
+		return currentReplicas, &utilization, utilizationInfo.OldestTimestamp, nil
 	}
 
-	return currentReplicas, &utilization, timestamp, nil
+	// for unready pods, in the case of scale up, check to see if treating those pods as having
+	// a mock CPU utilization of zero would change things
+	if utilizationInfo.UnreadyPodsCount > 0 && usageRatio > 1.0 {
+		// check if treating unready pods as having 0 CPU usage would change things
+		newUtilization := (utilizationInfo.Utilization * float64(utilizationInfo.ReadyPodsCount)) / float64(utilizationInfo.ReadyPodsCount+utilizationInfo.UnreadyPodsCount)
+		newUsageRatio := newUtilization / float64(targetUtilization)
+
+		// simply don't scale if the new usage ratio would mean a downscale or no scale
+		if newUsageRatio < 1.0 || math.Abs(1.0-newUsageRatio) <= tolerance {
+			return currentReplicas, &utilization, utilizationInfo.OldestTimestamp, nil
+		}
+
+		// set the usage ratio to scale by to our new usage ratio (which may be a bit more conservative)
+		usageRatio = newUsageRatio
+
+		// keep the reported utilization to be whatever we retrieved from the the ready pods
+	}
+
+	return int32(math.Ceil(usageRatio * float64(currentReplicas))), &utilization, utilizationInfo.OldestTimestamp, nil
```

Comment:
if we want to get really technical here, pending pods and unready pods should probably be counted the same, *but* non-pending non-running pods should be different (succeeded, failed, etc), because they'll never transition back to running.  I'll see if I can accurately reflect that